### PR TITLE
Fixes #22370: use the new RH repos page instead of the old.

### PR DESCRIPTION
**Pull request by waldenraines**
Also removes the old Red Hat Repository page code.

http://projects.theforeman.org/issues/22370

### CODE REVIEW

File: app/controllers/katello/products_controller.rb [hunk 1]
```diff
@@ -12,34 +12,6 @@ def section_id
...
-                       ::Actions::Katello::RepositorySet::EnableRepository
-                     else
-                       ::Actions::Katello::RepositorySet::DisableRepository
-                     end
-      task = sync_task(action_class, @product, @content, substitutions)
-      render :json => { :task_id => task.id }
-    rescue => e
-      render :partial => 'katello/providers/redhat/enable_errors', :locals => { :error_message => e.message}, :status => 500
-    end
-
```
Comment (jlsherrill) [inline app/controllers/katello/products_controller.rb]: I think this whole controller can be deleted

Comment (waldenraines) [reply app/controllers/katello/products_controller.rb]: Nice, thanks, you know how I love deleting stuff.

**Comment by waldenraines** [comment]
Thoughts on leaving the route for the old page and replacing the menu item with the new URL?  That would give us a workaround page if something comes up.

**Comment by jlsherrill** [comment]
@waldenraines i'm not sure i understand your question?

**Comment by waldenraines** [comment]
> @waldenraines i'm not sure i understand your question?

Instead of deleting the files leaving them for this release just in case we need a workaround for enabling repos but changing the menu item to point to the new url (/redhat_repositories).  The old url (/katello/providers/redhat) would still exist but just not be linked from anywhere.

**Comment by jlsherrill** [comment]
I think if there is some problem that is not obvious now, it should be obscure enough and able to be worked around with the cli.  I'd go for ditching the old page entirely at this point.

**Comment by waldenraines** [comment]
> I think if there is some problem that is not obvious now, it should be obscure enough and able to be worked around with the cli. I'd go for ditching the old page entirely at this point.

Works for me.

File: app/controllers/katello/products_controller.rb (hunk 1 above)
Comment (waldenraines) [reply app/controllers/katello/products_controller.rb]: @jlsherrill updated.

**Comment by jturel** [comment]
Heard you like deleting code - I think app/helpers/katello/providers_helper.rb is another candidate

**Comment by jturel** [comment]
From the /xui page "Click on the kebab to the right to navigate through the available pages." - however I see no kebab

File: lib/katello/plugin.rb [hunk 2]
```diff
@@ -19,8 +19,9 @@
     menu :top_menu,
          :redhat_provider,
          :caption => N_('Red Hat Repositories'),
-         :url_hash => {:controller => 'katello/providers',
```
Comment (jturel) [inline lib/katello/plugin.rb]: Thoughts on providing a redirect to the new page from katello/providers/redhat_provider? It's become a 404 now

**Comment by waldenraines** [comment]
> From the /xui page "Click on the kebab to the right to navigate through the available pages." - however I see no kebab

That page should be gone completely, I removed it with the latest push.

File: lib/katello/plugin.rb (hunk 2 above)
Comment (waldenraines) [reply lib/katello/plugin.rb]: That's a good suggestion, added.

File: config/routes.rb [hunk 3]
```diff
@@ -10,25 +10,14 @@
...
-        get :redhat_provider_tab
-      end
-    end
-
     if Katello.with_remote_execution?
       match '/remote_execution' => 'remote_execution#create', :via => [:post]
     end
   end
 
+  get '/katello/providers/redhat_provider', to: redirect('/subscriptions')
```
Comment (jturel) [inline config/routes.rb]: Is was expecting redirect from old RH repos page (redhat_provider) to new RH repos page (/redhat_repositories)

File: config/routes.rb [hunk 4]
```diff
@@ -10,25 +10,14 @@
...
-    resources :products, :only => [] do
-      member do
-        get :available_repositories
-        put :toggle_repository
-      end
-    end
-
-    resources :providers, :only => [] do
-      collection do
-        get :redhat_provider
```
Comment (jturel) [inline config/routes.rb]: There's a path helper referring to this route that should be updated unless that page is also dead code:

https://github.com/Katello/katello/blob/ffe498835c7ee2c6aa9795aee6292202453bb92f/app/views/katello/sync_management/_products.html.erb#L34

File: config/routes.rb (hunk 3 above)
Comment (waldenraines) [reply config/routes.rb]: Lol, oops, too many things going on.  Apparently the RH subscriptions page is on my mind.  Will fix.

File: config/routes.rb (hunk 4 above)
Comment (waldenraines) [reply config/routes.rb]: Good catch, will fix.

**Comment by waldenraines** [comment]
@jturel updated.

**Comment by jturel** [comment]
Looks good walden, APJ

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test katello]

**Pull request merged**